beacon_node/network/src/subnet_service/mod.rs

```diff
@@ -128,20 +129,33 @@ impl<T: BeaconChainTypes> SubnetService<T> {
 
         // Build the list of known permanent subscriptions, so that we know not to subscribe or
         // discover them.
-        let mut permanent_attestation_subscriptions = HashSet::default();
+        let mut permanent_attestation_subscriptions = BitVector::new();
         if config.subscribe_all_subnets {
             // We are subscribed to all subnets, set all the bits to true.
             for index in 0..beacon_chain.spec.attestation_subnet_count {
-                permanent_attestation_subscriptions
-                    .insert(Subnet::Attestation(SubnetId::from(index)));
+                let index_usize = index as usize;
+                if let Err(e) = permanent_attestation_subscriptions.set(index_usize, true) {
```

Comment:
This relies on `spec.attestation_subnet_count` (a runtime config) matching `EthSpec::SubnetBitFieldLength` (a compile time constant), which is a hard requirement, we could just fail the startup if this happens?